draft-tjhai-ipsecme-hybrid-qske-ikev2-04.xml

```diff
@@ -0,0 +1,994 @@
+                from the CREATE_CHILD_SA; KE(1)..KE(n), Ni(1)..Ni(n), Nr(1)..Nr(n) - shared keys and nonces
+                from additional key exchanges.
+                </t>
+
+            </section>
+        </section>
+    </section>
+
+    <section title="IANA Considerations" anchor="IANA">
+        <t>This document renames "Transform Type 4 - Diffie-Hellman Group Transform IDs" to "Transform Type 4 - Key Exchange Method Transform IDs"</t>
+
+        <t>This document also adds the following Transform Types to the "Transform Type Values" registry:</t>
+        <figure align="left"><artwork align="left"><![CDATA[
+Type Description                Used In                        Reference
+------------------------------------------------------------------------
+6    Additional Key Exchange 1  (optional in IKE, AH and ESP)  [RFCXXXX]
+7    Additional Key Exchange 2  (optional in IKE, AH and ESP)  [RFCXXXX]
+8    Additional Key Exchange 3  (optional in IKE, AH and ESP)  [RFCXXXX]
+9    Additional Key Exchange 4  (optional in IKE, AH and ESP)  [RFCXXXX]
+10   Additional Key Exchange 5  (optional in IKE, AH and ESP)  [RFCXXXX]
+11   Additional Key Exchange 6  (optional in IKE, AH and ESP)  [RFCXXXX]
+12   Additional Key Exchange 7  (optional in IKE, AH and ESP)  [RFCXXXX]
+        ]]></artwork></figure>
+
+        <t>This document also defines a new Notify Message Type in the "Notify Message Types - Status Types" registry:</t>
+
+        <figure align="center"><artwork align="left"><![CDATA[
+<TBA>       ADDITIONAL_KEY_EXCHANGE
+        ]]></artwork></figure>
+
+        <t>and a new Notify Message Type in the "Notify Message Types - Error Types" registry:</t>
+
+        <figure align="center"><artwork align="left"><![CDATA[
+<TBA>       STATE_NOT_FOUND
+        ]]></artwork></figure>
+    </section>
+
+	<section title="Security Considerations" anchor="security">
+        <t>
+    	The key length of the Encryption Algorithm (Transform Type 1), the
+    	Pseudorandom Function (Transform Type 2) and the Integrity Algorithm
+    	(Transform Type 3), all have to be of sufficient length to prevent
+    	attacks using Grover's algorithm <xref target="GROVER"/>.  In order to use the
+    	extension proposed in this document, the key lengths of these
+    	transforms SHALL be at least 256 bits long in order to provide
+    	sufficient resistance to  quantum attacks.  Accordingly the
+    	post-quantum security level achieved is at least 128 bits.</t>
+    
+    	<t>SKEYSEED is calculated from shared, KEx, using an algorithm defined
+    	in Transform Type 2.  While a quantum attacker may learn the value
+    	of KEx', if this value is obtained by means of a classical key exchange,
+    	other KEx values generated by means of a quantum-resistant algorithm
+    	ensure that the final SKEYSEED is not compromised.  This assumes that
+    	the algorithm defined in the Transform Type 2 is post-quantum.
+    	</t>
```

Comment:
I don't we ever refer to KEx and KEx' anymore, so we should re-phrase this paragraph.

```
SKEYSEED is calculated from a set of shared secrets using an algorithm defined in
Transform Type 2.  While a quantum attacker may learn the value of the shared
secret exchange during the IKE_SA_INIT or CREATE_CHILD_SA phase, if this value is
obtained by means of a classical key exchange, other shared secrets generated
by means of quantum-resistant algorithms ensure that the final SKEYSEED value
is not compromised.  This assumes that the algorithm defined in the Transform
Type 2 is post-quantum secure.
```

Reply:
I'm OK with this text.